src/backend/distributed/operations/shard_split.c

```diff
@@ -719,23 +694,32 @@ DoSplitCopy(WorkerNode *sourceShardNode, List *sourceColocatedShardIntervalList,
 	forboth_ptr(sourceShardIntervalToCopy, sourceColocatedShardIntervalList,
 				splitShardIntervalList, shardGroupSplitIntervalListList)
 	{
-		StringInfo splitCopyUdfCommand = CreateSplitCopyCommand(sourceShardIntervalToCopy,
-																splitShardIntervalList,
-																destinationWorkerNodesList);
+		/*
+		 * Skip copying data for partitioned tables, because they contain no
+		 * data themselves. Their partitions do contain data, but those are
+		 * different colocated shards that will be copied seperately.
+		 */
+		if (!PartitionedTable(sourceShardIntervalToCopy->relationId))
+		{
+			StringInfo splitCopyUdfCommand = CreateSplitCopyCommand(
+				sourceShardIntervalToCopy,
+				splitShardIntervalList,
+				destinationWorkerNodesList);
 
-		Task *splitCopyTask = CreateBasicTask(
-			sourceShardIntervalToCopy->shardId, /* jobId */
-			taskId,
-			READ_TASK,
-			splitCopyUdfCommand->data);
+			Task *splitCopyTask = CreateBasicTask(
+				sourceShardIntervalToCopy->shardId, /* jobId */
```

Comment:
this should probably be INVALID_JOB_ID

Reply:
In general I was using a valid jobid hopping we log some traces per jobid that can be helpful for debugging any potential issues in prod.